Automattic/woocommerce-payments
# Fix WP & WC versions for E2E Tests + Optimizations

Fixes #3554 

#### Changes proposed in this Pull Request

At the moment, the E2E setup script doesn't take into account WordPress & WooCommerce versions specified in GitHub Actions workflow matrix. This PR aims to fix the same along with some optimizations to the setup script. Here are the important changes.

* Allow setup script to pick the specified WordPress & WooCommerce versions (both on GitHub Actions & Local)
* Fixes problems with docker mount points
* Optimizes downloading the latest versions for WooCommerce Subscriptions, Action Scheduler & WooCommerce Gutenberg Product Blocks
* Adds dependency check for the setup script since the script uses the `jq` library which doesn't come pre-installed.
* Adds E2E environment config logging to confirm whether it uses the specified WP & WC versions
* Adds new bash function to show output from WP-CLI commands even if `DEBUG` flag is not present or set to false.

#### Testing instructions

* Run E2E tests manually against this branch from the Actions tab. Check the `Prepare test environment` section to see the running configuration. Confirm that the WP, WC & PHP versions match the ones specified in GH Actions.
* On local, run the E2E setup after defining the `local.env` file based on the Readme. Try specifying WP & WC versions and confirm that the environment is setup correctly.

-------------------


**Post merge**
